x-pack/solutions/observability/plugins/apm/public/components/app/error_group_details/error_sampler/error_sample_agent_builder_ai_insight.tsx

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import dedent from 'dedent';
+import { i18n } from '@kbn/i18n';
+import React, { useMemo, useState } from 'react';
+import { EuiSpacer } from '@elastic/eui';
+import type { AT_TIMESTAMP } from '@kbn/apm-types';
+import { AiInsight } from '@kbn/observability-agent-builder';
+import { OBSERVABILITY_AI_INSIGHT_ATTACHMENT_TYPE_ID } from '../../../../../common/agent_builder/attachment_ids';
+import { useApmPluginContext } from '../../../../context/apm_plugin/use_apm_plugin_context';
+import type { APMError } from '../../../../../typings/es_schemas/ui/apm_error';
+import { getIsObservabilityAgentEnabled } from '../../../../../common/agent_builder/get_is_obs_agent_enabled';
+import { useAnyOfApmParams } from '../../../../hooks/use_apm_params';
+import { useTimeRange } from '../../../../hooks/use_time_range';
+import { callApmApi } from '../../../../services/rest/create_call_apm_api';
+import { useLocalStorage } from '../../../../hooks/use_local_storage';
+
+export function ErrorSampleAgentBuilderAiInsight({
+  error,
+  transaction,
+}: {
+  error: {
+    [AT_TIMESTAMP]: string;
+    error: Pick<APMError['error'], 'log' | 'exception' | 'id'>;
+    service: {
+      name: string;
+      environment?: string;
+      language?: {
+        name?: string;
+      };
+      runtime?: {
+        name?: string;
+        version?: string;
+      };
+    };
+  };
+  transaction?: {
+    transaction: {
+      name: string;
+    };
+  };
+}) {
+  const { onechat, core, inference } = useApmPluginContext();
+  const isObservabilityAgentEnabled = getIsObservabilityAgentEnabled(core);
+
+  const { query } = useAnyOfApmParams(
+    '/services/{serviceName}/errors/{groupId}',
+    '/mobile-services/{serviceName}/errors-and-crashes/errors/{groupId}',
+    '/mobile-services/{serviceName}/errors-and-crashes/crashes/{groupId}'
+  );
+  const { rangeFrom, rangeTo, environment, kuery } = query;
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [summary, setSummary] = useState('');
+  const [context, setContext] = useState('');
+
+  const [lastUsedConnectorId] = useLocalStorage('agentBuilder.lastUsedConnector', '');
```

Comment:
Can you use this hook exposed by AB? https://github.com/elastic/kibana/blob/2d666cad2c5d0b3ae6f9c28706b3efd0fdc75504/x-pack/platform/plugins/shared/onechat/public/application/hooks/chat/use_connector_selection.ts#L20

Reply:
This way we do not duplicate the locale storage key